tico/experimental/quantization/ptq/examples/quantize_with_gptq.py

```diff
@@ -0,0 +1,161 @@
+from tico.experimental.quantization.ptq.wrappers.ptq_wrapper import PTQWrapper
+from tico.experimental.quantization.ptq.wrappers.quant_module_base import (
+    QuantModuleBase,
+)
+
+# -------------------------------------------------------------------------
+# 0. Global configuration
+# -------------------------------------------------------------------------
+MODEL_NAME = "meta-llama/Meta-Llama-3-1B"
+DEVICE = "cuda" if torch.cuda.is_available() else "cpu"
+STRIDE = 512
+
+# Token-budget presets for calibration / evaluation.
+TOKENS: dict[str, int] = {
+    # Smoke test (<1 min turnaround on CPU/GPU)
+    "debug": 2_000,  # ≈16 × 128-seq batches
+    # Good default for 1-7B models (≲3 % ppl delta)
+    "baseline": 50_000,
+    # Production / 4-bit observer smoothing
+    "production": 200_000,
+}
+CALIB_TOKENS = TOKENS["baseline"]
+
+# -------------------------------------------------------------------------
+# 1. Helper — copy GPTQ (scale, zp) into PTQ observers
+# -------------------------------------------------------------------------
+def inject_gptq_qparams(
+    root: torch.nn.Module,
+    gptq_quantizers: dict[str, Any],  # {fp_name: quantizer}
+    weight_obs_name: str = "weight",
+):
+    """
+    For every QuantModuleBase whose `fp_name` matches a GPTQ key,
+    locate the observer called `weight_obs_name` and overwrite its
+    (scale, zero-point), then *lock* them against further updates.
+    """
+    for m in root.modules():
+        if not isinstance(m, QuantModuleBase):
+            continue
+        if m.fp_name is None:
+            continue
+        quantizer = gptq_quantizers.get(m.fp_name)
+        if quantizer is None:
+            continue
+        obs = m.get_observer(weight_obs_name)
+        if obs is None:
+            continue
+        # GPTQ quantizer attributes
+        obs.load_qparams(quantizer.scale, quantizer.zero, lock=True)
+
+
+# -------------------------------------------------------------------------
+# 2. Load the FP backbone
+# -------------------------------------------------------------------------
+print("Loading FP model …")
+tokenizer = AutoTokenizer.from_pretrained(MODEL_NAME)
+model = AutoModelForCausalLM.from_pretrained(MODEL_NAME).to(DEVICE).eval()
+model.config.use_cache = False  # disable KV-cache → full forward
+m_to_fqn = build_fqn_map(model)  # map modules → fully-qualified names
+
+# -------------------------------------------------------------------------
+# 3. Run GPTQ (weight-only) pass
+# -------------------------------------------------------------------------
+print("Applying GPTQ …")
+dataset = load_dataset("wikiText", "wikitext-2-raw-v1", split="test")
+q_m = prepare(model, GPTQConfig(), inplace=True)
+
+for d in tqdm.tqdm(dataset, desc="GPTQ calibration"):
+    ids = tokenizer(d["text"], return_tensors="pt").input_ids.to(DEVICE)
+    q_m(ids)  # observers gather weight stats
+
+q_m = convert(q_m, inplace=True)  # materialize INT-weight tensors
```

Comment:
It's great use of tqdm above but when running this script, `q_m = convert(q_m, inplace=True)` takes more time.
Could you add tqdm to quantizer.py too?

```py
        quantizers: Dict[str, Any] = {}
        for l_idx, layer in enumerate(tqdm.tqdm(target_layers)): # <-- to track the conversion in terminal.
```

Reply:
Yeah, I think so. I kept thinking about it and ketp putting it off, lol. Thanks for the reivew:)